src/webgpu/gpu_test.ts

```diff
@@ -167,6 +167,35 @@ export class GPUTest extends Fixture {
     });
   }
 
+  expectContentsMultipleValidValues(
+    src: GPUBuffer,
+    expected1: TypedArrayBufferView,
```

Comment:
If anything, I think this should take a `sequence<TypedArrayBufferView>`. Maybe we could also refactor the `expectContents` to take a validator function so it's easy to build additional variations in the future.

Reply:
Thanks for your suggestions, Corentin. I considered about this too but didn't do this because: 1) it is somehow over-engineering. 2) actually we have two directions to expand this function: 
- expand two values to an array (or a sequence), and the expected result should fall into any value of the array. it is just like any() function. This direction is what you said. 
- expand two values to two vectors, and the expected result should be a vector whose value at any index should be either  of the two values at the same index of the two vectors. Take (R, G, B, A) as an example, if two vectors are (0, 0, 0, 0) and (255, 255, 255, 255), then there are 16 correct values like (0, 0, 0, 0), (0, 0, 0, 255), ..., (0, 255, 255, 0), ..., (255, 255, 255, 255). This is another possible expansion. If we have multiple values in the buffer, a race condition of write after write is more likely to behave in this direction. But I don't want to make the test too complicated. 

So I didn't expand this function. 

How about we keep it and expand this function to whatever we want if we do need it in future?

Reply:
While I think it is ok in this test fixture, the method I have concerns keeping the method in the harness as-is: it says 'multiple values" but only take two. In general the harness needs to be higher quality and more future proof than test code.

What we could do is do the refactor of `expectContents` to to have a version that takes a validator and make the validator check agains the two values in the fixture.

Reply:
Yeah. I have concerns about the naming stuff too (I said about the naming problem at the very beginning). I hesitated to pick up a name like alternative, either, two, multiple, etc. Let's be clear and just say "two" valid values here. And we can expand it to multiple valid values or two mixed vectors if needed in future. I understand that the code should be future-proof but I don't think it is very clear that which direction is the future (maybe neither of them is needed). So I added a comment about it. WDYT?

Reply:
The comment is inconsistent with the idea of passing a validator function, but it links to this discussion, so it's fine. I also considered suggesting passing an array here (instead of 2 values); it wouldn't really be over-engineering as it wouldn't be any more complex. However I think a validator function makes more sense in the future. A validator function pattern also works much better for texture content expectation. (I was thinking of passing a `function(x,y,z)->expected value` into expectTextureContents but a validator is more flexible.)